sauna-app/components/map.tsx

```diff
@@ -0,0 +1,61 @@
+import 'leaflet/dist/leaflet.css';
+import { MapContainer,  TileLayer, Marker, Popup} from 'react-leaflet'
+import L from 'leaflet'
+import style from '../styles/Home.module.css'
+import { useState, useEffect } from 'react';
+import { SaunaData } from '../pages/typeDefinitions'
+
+
+
+function Map() {
+
+  const [saunas, setSaunas] = useState<SaunaData>();
+
+  useEffect(() => {
+    const fetchAPI = async () => {
+      await fetch('http://localhost:3000/api/saunas')
+      .then((response) => response.json())
+      .then((data:SaunaData) => setSaunas(data))
+    }
+    fetchAPI()
+  }, [])
+
+
+  return (
+    <MapContainer className={style.map} center={[60.257215, 24.966394]} zoom={10} scrollWheelZoom={false}>
+      <TileLayer
+        attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
+        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+      />
+      {saunas?.saunas.map(s => {
+        let latitude: number = s.latitude
+        let longitude: number = s.longitude
+        let name: string = s.name
+        let id: string =s.id
+        return (
+          <Marker 
+            key={id}
+            position={[latitude, longitude]}
+            eventHandlers={{
+              mouseover: (event) => event.target.openPopup(),
```

Comment:
From accessibility perspective, you probably want to add a focus event listener too to open the popup. But I'm fine if that's done in a separate PR :)

Reply:
I actually changed the Popup tags to Tooltip tags because they have the hovering effects on them so less code, but I have to read/test more how they work with focusing.